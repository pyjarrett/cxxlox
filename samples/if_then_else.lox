if (true) {
    print "lonely if: true";
}

if (true) {
    print "if: true";
}
else {
    print "Compiler error";
}

if (false) {
    print "Compiler error";
}
else {
    print "else: false";
}

if (true or false) {
    print "true or false";
}

if (false or true) {
    print "false or true";
}

if (true and false) {
    print "Compiler error";
}
else {
    print "else: true and false";
}

if (false and true) {
    print "Compiler error";
}
else {
    print "else: false and true";
}
