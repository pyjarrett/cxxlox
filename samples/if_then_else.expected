lonely if: true
if: true
else: false
true or false
false or true
else: true and false
else: false and true
